goodscary/firstrubyfriend
# Create user model

#### What's this PR do?

- Adds authentication-zero gem
- Adds rails/ulid.
- Adds Github login

##### Background context

Also tackles issue https://github.com/goodscary/firstrubyfriend/issues/4 here. Probably should have created a separate PR, I initially wanted to have a single migration file and then was too far into implementing it to back out :)

#### Where should the reviewer start?

- ULID implementation. I set the migration files as per the documentation and should be correct. Their doc was for mysql but its mostly the same for postgres.
- Authentication zero. Couldn't figure out how to get Current.user so I added the `  before_action :set_current_user` to ApplicationController. This is then used in the home#show page.


#### How should this be manually tested?

- Attempt to sign up and sign in using email + password.
-  Attempt to sign up and sign in using Github login.
- Once a user and sessions are created, verify if the ULID is working and stored in the database correctly.


#### Screenshots or Video
https://www.loom.com/share/59433a57d2c64103ac1eb131a0fbc276


---

#### Migrations

- User model with first_name, last_name, email, password
- Models added by authentication-zero: session, password_reset_token, email_verifitcation_token

#### Additional deployment instructions

- We need to add Github app credentials.


#### Additional ENV Vars

I re-rolled the rails credentials as it wouldn't decrypt.
